pallets/nft/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+            ticker,
+            collection_id,
+        ));
+        Ok(())
+    }
+
+    fn base_mint_nft(
+        origin: T::Origin,
+        ticker: Ticker,
+        metadata_attributes: Vec<NFTMetadataAttribute>,
+    ) -> DispatchResult {
+        // Verifies if the collection exists
+        let collection_id =
+            CollectionTicker::try_get(&ticker).map_err(|_| Error::<T>::CollectionNotFound)?;
+
+        // Verifies if the caller has the right permissions (regarding asset and portfolio)
+        let caller_portfolio = Asset::<T>::ensure_agent_with_custody_and_perms(
+            origin,
+            ticker.clone(),
+            PortfolioKind::Default,
+        )?;
+
+        // Verifies that all mandatory keys are being set and that there are no duplicated keys
+        let mandatory_keys: BTreeSet<AssetMetadataKey> = Self::collection_keys(&collection_id);
+        ensure!(
+            mandatory_keys.len() == metadata_attributes.len(),
+            Error::<T>::InvalidMetadataAttribute
+        );
+
+        let n_keys = metadata_attributes.len();
+        let nft_attributes: BTreeMap<_, _> = metadata_attributes
+            .into_iter()
+            .map(|a| (a.key, a.value))
+            .collect();
+        ensure!(
+            n_keys == nft_attributes.len(),
+            Error::<T>::DuplicateMetadataKey
+        );
+
+        for metadata_key in nft_attributes.keys() {
+            ensure!(
+                mandatory_keys.contains(metadata_key),
+                Error::<T>::InvalidMetadataAttribute
+            );
+        }
+
+        // Mints the NFT and adds it to the caller's portfolio
+        let new_balance = BalanceOf::get(&ticker, &caller_portfolio.did)
+            .checked_add(ONE_UNIT)
+            .ok_or(Error::<T>::BalanceOverflow)?;
+        let nft_id = NextNFTId::try_mutate(&collection_id, try_next_pre::<T, _>)?;
+        BalanceOf::insert(&ticker, &caller_portfolio.did, new_balance);
+        for (metadata_key, metadata_value) in nft_attributes.into_iter() {
+            MetadataValue::insert((&collection_id, &nft_id), metadata_key, metadata_value);
+        }
+        Portfolio::<T>::add_portfolio_nft(caller_portfolio.did, collection_id, nft_id);
+
+        Self::deposit_event(Event::MintedNft(
+            caller_portfolio.did,
+            collection_id,
+            nft_id,
+        ));
+        Ok(())
+    }
+
+    fn base_burn_nft(
+        origin: T::Origin,
+        ticker: Ticker,
+        nft_id: NFTId,
+        portfolio_kind: PortfolioKind,
+    ) -> DispatchResult {
+        // Verifies if the collection exists
+        let collection_id =
+            CollectionTicker::try_get(&ticker).map_err(|_| Error::<T>::CollectionNotFound)?;
+
+        // Ensure origin is agent with custody and permissions for default portfolio.
+        let caller_portfolio = Asset::<T>::ensure_agent_with_custody_and_perms(
+            origin,
+            ticker,
+            PortfolioKind::Default,
```

Comment:
Since the call can burn from a non-default portfolio, the permissions and custody check needs to be done on the same portfolio.
```suggestion
            portfolio_kind,
```